core/field.ts

```diff
@@ -957,49 +964,57 @@ export abstract class Field<T = unknown> implements IASTNodeLocationSvg,
    * @param newValue New value.
    * @sealed
    */
-  setValue(newValue: AnyDuringMigration) {
+  setValue(newValue: T|null) {
     const doLogging = false;
     if (newValue === null) {
       doLogging && console.log('null, return');
       // Not a valid value to check.
       return;
     }
 
-    let validatedValue = this.doClassValidation_(newValue);
-    // Class validators might accidentally forget to return, we'll ignore that.
-    newValue = this.processValidation_(newValue, validatedValue);
-    if (newValue instanceof Error) {
+    const classValidatedValue = ((newValue: T): T|Error => {
+      const validatedValue = this.doClassValidation_(newValue);
+      // Class validators might accidentally forget to return, we'll ignore
+      // that.
+      return this.processValidation_(newValue, validatedValue);
+    })(newValue);
```

Comment:
On the note of logging - is it necessary to leave it in? Perhaps it's there since this is common to debug, though it's just a constant defined in scope, so it will never log at run time unless somebody manually updates `core/field.ts` locally.

Minor implementation tweak - just need to add `|| null`
```ts
const localValidatedValue = this.processValidation_(
        classValidatedValue,
        this.getValidator()?.call(this, classValidatedValue) || null);
```

Overall, that approach would work for me if it works for @maribethb! Might be a tad busy for one liners, though I think they're still generally simple enough to follow. They solve what I was trying to address with the IIFE as well.